app/controllers/deploys_controller.rb

```diff
@@ -1,18 +1,13 @@
 class DeploysController < ApplicationController
-  include CurrentProject
   include ProjectLevelAuthorization
 
-
-  before_action except: [:active, :active_count, :recent, :changeset] do
-    find_project(params[:project_id])
-  end
-
```

Comment:
> If you look in the routes.rb file, you'll find a lot of controllers are defined inside of the resources :project block. That means that those endpoints should only ever get called if a valid project is specified.

Yep, noticed that.

> That way if there's an invalid project, that is detected immediately, rather than at some random point wherever current_project is first called.
(...)
What do you think of that idea?

I understand your point about failing at some random point when `current_project` is first used, but I still have some doubts in here. When PR [#553](https://github.com/zendesk/samson/pull/553) was first reviewed, and the guys "complained" about the way this before_actions were being used, I started googling for it, and some of the first hits I got kind of explain why should we avoid using before_action callbacks for loading data before actions get executed, and most of the arguments actually make sense, specially the arguments about:
- Sacrificing clarity for the sake of DRY-ness (authorisation filters depending on the `find_project` filter follows the same example described in the articles). However, we already got rid of this by introducing the `current_project` method.
- Filters coupled with controller actions (actions logic, and view templates, depending on the execution of the `find_project` method).
- Changes to the controller's state before the action gets executed (but this will happen anyway when the `current_project` method gets executed by the authorisation filters)

You can find those references in here:
[Don't use before_action to load data](http://craftingruby.com/posts/2015/05/31/dont-use-before-action-to-load-data.html)
[Better alternative to Rails' before_action to load objects](http://jgaskins.org/blog/2014/08/25/better-alternative-to-rails-before_action-to-load-objects)

This being said, I know articles like this can be a little opinionated and you have much more experience with Rails than me. So, if you believe that's the way to go, then your suggestion might be the best one.

What's your opinion on these articles?